test/net/sourceforge/kolmafia/request/AscensionRequestTest.java

```diff
@@ -0,0 +1,143 @@
+package net.sourceforge.kolmafia.request;
+
+import static internal.helpers.Networking.html;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.RequestLogger;
+import net.sourceforge.kolmafia.preferences.Preferences;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class AscensionRequestTest {
+  @BeforeEach
+  public void initializeCharPrefs() {
+    KoLCharacter.reset("fakePrefUser");
+    KoLCharacter.reset(true);
+    Preferences.saveSettingsToFile = false;
+  }
+
+  @AfterEach
+  public void resetCharAndPrefs() {
+    KoLCharacter.reset("");
+    KoLCharacter.reset(true);
+    KoLCharacter.setUserId(0);
+    Preferences.saveSettingsToFile = false;
+  }
+
+  @Test
+  void testAscensionsTodayTracked() {
+    // Simple test to confirm that it resets our last run breakfast
+    String breakfastPref = "lastBreakfast";
+    int breakfastBefore = 5;
+    int breakfastAfter = 0;
+    int breakfastValhalla = -1;
+    String ascensionsPref = "ascensionsToday";
+    int ascensionBefore = 1;
+    int ascensionAfter = 2;
+
+    // Set breakfast and ascension prefs
+    Preferences.setInteger(ascensionsPref, ascensionBefore);
+    Preferences.setInteger(breakfastPref, breakfastBefore);
+
+    // This does a 302 redirect to afterlife.php
+    var jumpGash = new GenericRequest("ascend.php?action=ascend&pwd&confirm=on&confirm2=on");
+    jumpGash.execute();
+
+    // Confirm our last run breakfast is now 0
+    assertEquals(breakfastAfter, Preferences.getInteger(breakfastPref));
+    // Confirm our ascensions today has not changed
+    assertEquals(ascensionBefore, Preferences.getInteger(ascensionsPref));
+
+    // Execute the result of the 302 redirect and get the output which puts us in valhalla
+    var afterlife = new GenericRequest("afterlife.php");
```

Comment:
Please use fakeHttpClient and `run` here

Reply:
Done